backend/src/server/server.go

```diff
@@ -64,33 +68,52 @@ func utilityRoutes(router fiber.Router) {
 	})
 }
 
-func userRoutes(router fiber.Router, userService services.UserServiceInterface) {
+func userRoutes(router fiber.Router, userService services.UserServiceInterface, middlewareService middleware.MiddlewareInterface) {
 	userController := controllers.NewUserController(userService)
 
+	// api/v1/users/*
 	users := router.Group("/users")
-
 	users.Post("/", userController.CreateUser)
-	users.Get("/", userController.GetUsers)
-	users.Get("/:id", userController.GetUser)
-	users.Patch("/:id", userController.UpdateUser)
-	users.Delete("/:id", userController.DeleteUser)
-
-	userTags := users.Group("/:uid/tags")
-	
-	userTags.Post("/", userController.CreateUserTags)
-	userTags.Get("/", userController.GetUserTags)
+	users.Get("/", middlewareService.Authorize(types.UserReadAll), userController.GetUsers)
+
+	// api/v1/users/:id/*
+	usersID := users.Group("/:id")
+	usersID.Use(middlewareService.UserAuthorizeById)
+
+	usersID.Get("/", middlewareService.Authorize(types.UserRead), userController.GetUser)
+	usersID.Patch("/", middlewareService.Authorize(types.UserWrite), userController.UpdateUser)
+	usersID.Delete("/", middlewareService.Authorize(types.UserDelete), userController.DeleteUser)
+
+	usersID.Post("/tags", userController.CreateUserTags)
+	usersID.Get("/tags", userController.GetUserTags)
 }
 
-func clubRoutes(router fiber.Router, clubService services.ClubServiceInterface) {
+func clubRoutes(router fiber.Router, clubService services.ClubServiceInterface, middlewareService middleware.MiddlewareInterface) {
 	clubController := controllers.NewClubController(clubService)
 
 	clubs := router.Group("/clubs")
 
-	clubs.Get("/", clubController.GetAllClubs)
+	clubs.Get("/", middlewareService.Authorize(types.ClubReadAll), clubController.GetAllClubs)
 	clubs.Post("/", clubController.CreateClub)
-	clubs.Get("/:id", clubController.GetClub)
-	clubs.Patch("/:id", clubController.UpdateClub)
-	clubs.Delete("/:id", clubController.DeleteClub)
+
+	// api/v1/clubs/:id/*
+	clubsID := clubs.Group("/:id")
+	clubsID.Use(middlewareService.ClubAuthorizeById)
+
+	clubsID.Get("/", middlewareService.Authorize(types.ClubRead), clubController.GetClub) // TODO: Should you be able to get a club without being logged in? (yes)
```

Comment:
TODO resolved or should we open an issue / discussion about this then remove the comment?